File: backend/app/schemas/health.py
from pydantic import BaseModel, Field


class AIProviderStatus(BaseModel):
    provider: str
    configured: bool
    enabled: bool
    active_for_chat: bool
    active_for_embedding: bool
    ok: bool
    latency_ms: float | None = None
    model: str | None = None
    error: str | None = None


class AIStatusResponse(BaseModel):
    chat_provider: str
    embedding_provider: str
    hot_reload: bool
    providers: list[AIProviderStatus]


class HealthChecks(BaseModel):
    database: bool
    redis: bool
    resend: bool = False
    ai: bool
    ai_provider: str | None = None
    ai_latency_ms: float | None = None
    openai: bool = True
    openai_latency_ms: float | None = None


class HealthResponse(BaseModel):
    status: str = Field(description="healthy | degraded")
    checks: HealthChecks

File: frontend/src/types/health.ts
/** Health & AI provider probe API contracts. */

export type PlatformHealthStatus = "healthy" | "degraded";

export interface HealthChecks {
  database: boolean;
  redis: boolean;
  ai: boolean;
  ai_provider: string | null;
  ai_latency_ms: number | null;
  /** @deprecated use `ai` — kept for backward compatibility with older backends */
  openai: boolean;
  openai_latency_ms: number | null;
}

export interface HealthResponse {
  status: PlatformHealthStatus | string;
  checks: HealthChecks;
}

export interface AIProviderStatus {
  provider: string;
  configured: boolean;
  enabled: boolean;
  active_for_chat: boolean;
  active_for_embedding: boolean;
  ok: boolean;
  latency_ms: number | null;
  model: string | null;
  error: string | null;
}

export interface AIStatusResponse {
  chat_provider: string;
  embedding_provider: string;
  hot_reload: boolean;
  providers: AIProviderStatus[];
}
